src/components/Coupon/index.jsx

```diff
@@ -45,12 +45,25 @@ class Coupon extends React.Component {
     }
   }
 
+  handleCouponKeyDown(event) {
+    const { isExpanded } = this.state;
+    if (!isExpanded && isTriggerKey({ triggerKeys, action: 'OPEN_DETAILS', key: event.key })) {
+      event.preventDefault();
+      this.toggleCouponDetails();
+    } else if (isExpanded && isTriggerKey({ triggerKeys, action: 'CLOSE_DETAILS', key: event.key })) {
+      event.preventDefault();
+      this.toggleCouponDetails();
+    }
+  }
+
+  // eslint-disable-next-line react/no-unused-class-component-methods
   setCouponOpacity(dimmedStatus) {
     this.setState({
       dimmed: dimmedStatus,
     });
   }
 
+  // eslint-disable-next-line react/no-unused-class-component-methods
```

Comment:
nit: Is it worth adding a comment here explaining why we need these class component methods to remain even though they're not directly used by this class component?

Reply:
`setCouponOpacity` is being engaged in `ManageCodesTab` in `setCouponOpacity` method there.
I couldn't find a usage for closeCouponDetails but there is a test for it (sets state correctly on closeCouponDetails()) which fails on removal, the test could be removed along with the method.